src/components/common/pagination.svelte

```diff
@@ -0,0 +1,168 @@
+<script>
+  import { createEventDispatcher } from 'svelte';
+  import Button from 'ui/button.svelte';
+  import TextField from 'ui/text-field.svelte';
+
+  export let pages;
+  export let currentPage = 1;
+
+  if (currentPage < 1 || currentPage > pages) {
+    currentPage = 1;
+  }
+
+  let leftInputActive = false;
+  let rightInputActive = false;
+
+  const leftEllipsisKey = {};
+  const rightEllipsisKey = {};
+
+  function generateButtonList() {
+    let list = [];
+    if (currentPage !== 1) {
+      list.push(1);
+    }
+
+    if (currentPage - 1 == 3) {
+      list.push(currentPage - 2);
+    } else if (currentPage - 1 > 3) {
```

Comment:
Why all the weird arithmetics? These can simply be
`currentPage == 4`
`list.push(2)`
`currentPage > 4`

Reply:
Merely for symmetry with the other edge, `pages`, where such simplifications cannot be made.